### PR TITLE
#109: Make maximum and default grades configurable.

### PR DESCRIPTION
**Pull request by mattflax**
This PR makes the following changes, as per this comment on the issue: https://github.com/SeaseLtd/rated-ranking-evaluator/issues/109#issuecomment-598304539

- Adds the maximum and default (missing) grades to the Maven configuration for the four evaluation plugins, with defaults of 3 and 2 respectively (as per current behaviour);
- Adds maximum and default grades to constructors for ExpectedReciprocalRank, ReciprocalRank, NDCGAtK;
- Makes the metric names for ERR, RR, NDCG@K configurable in the parameterized metric configuration, with current values as defaults;
- Adds a singleton class for instantiating the MetricClassManager instance, and holding the configured max and missing grades specified in the pom.xml - this simplifies the MetricClassManager creation, and makes the grades available from anywhere.

Please let me know of any feedback on these changes!

### CODE REVIEW

File: rre-core/src/main/java/io/sease/rre/core/domain/metrics/MetricClassConfigurationManager.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    private static MetricClassConfigurationManager instance;
+
+    private BigDecimal defaultMaximumGrade = BigDecimal.valueOf(3);
+    private BigDecimal defaultMissingGrade = BigDecimal.valueOf(2);
+
+    private MetricClassConfigurationManager() {
+        // Private constructor
+    }
+
+    public static MetricClassConfigurationManager getInstance() {
```
Comment (agazzarini) [inline rre-core/src/main/java/io/sease/rre/core/domain/metrics/MetricClassConfigurationManager.java]: The constructor is empty so we could do 

`private final static MetricClassConfigurationManager INSTANCE = new MetricClassConfigurationManager();`

what do you think?

**Comment by agazzarini** [comment]
Great stuff @mattflax (or @mattoscln ?), I put just one trivial comment.

File: rre-core/src/main/java/io/sease/rre/core/domain/metrics/MetricClassConfigurationManager.java (hunk 1 above)
Comment (mattflax) [reply rre-core/src/main/java/io/sease/rre/core/domain/metrics/MetricClassConfigurationManager.java]: Sure, I've just made that change. My thinking was to not create before it is required, but it is used at the very beginning, and is not a big class either, so it's a sensible change to make.

**Comment by mattflax** [comment]
Thanks @agazzarini - I've responded above.

(It's @mattflax btw - any commits from the other account were by accident!)

**Comment by epugh** [comment]
Hi @agazzarini hope things are going well!  I'm writing up details for deploying RRE, and wanted to include the changes created by this PR in the documentation.  Wanted to see if there was anything I or @mattflax can do to help get this reviewed and merged?    I know everything is crazy!   Very happy to update the Wiki once this is merged for you ;-)

**Comment by agazzarini** [comment]
Hi @epugh many thanks, everything is doing fine here, even we are living in a surreal context. Hope the same there. 
I'll check and merge this PR tomorrow morning, and about the Wiki, feel free to add update.

Thanks again for everything

**Comment by agazzarini** [comment]
@mattflax @epugh: merged, thanks again

**Pull request merged**